src/main/webapp/app/course/competencies/competency-form/competency-form.component.ts

```diff
@@ -160,15 +164,15 @@ export class CompetencyFormComponent implements OnInit, OnChanges {
         if (this.isEditMode && this.formData && this.formData.title) {
             initialTitle = this.formData.title;
         }
-        this.form = this.fb.group({
+        this.form = this.fb.nonNullable.group({
             title: [
                 undefined as string | undefined,
                 [Validators.required, Validators.maxLength(255)],
                 [this.titleUniqueValidator(this.competencyService, this.courseId, initialTitle)],
             ],
             description: [undefined as string | undefined, [Validators.maxLength(10000)]],
             softDueDate: [undefined],
-            taxonomy: [undefined, [Validators.pattern('^(' + Object.keys(this.competencyTaxonomy).join('|') + ')$')]],
+            taxonomy: [undefined as CompetencyTaxonomy | undefined],
```

Comment:
This looks strange, can you explain this? Especially the `| undefined` part?

Reply:
undefined is an acceptable value, as a competency does not 100% need a competency.
The cast is used to create a form control of the type `Competency Taxonomy | undefined` with the formBuilder.